src/main/kotlin/org/octopusden/octopus/oc/template/plugins/gradle/service/diagnostics/DiagnosticsCollector.kt

```
@@ -0,0 +1,403 @@
+            return try {
+                val pb = ProcessBuilder(listOf("oc") + args).redirectErrorStream(false)
+                val p = pb.start()
+                val finished = p.waitFor(timeoutMs, TimeUnit.MILLISECONDS)
+                if (!finished) {
+                    p.destroyForcibly()
+                    return OcResult(-1, "", "timeout after ${timeoutMs}ms")
+                }
+                val out = p.inputStream.bufferedReader().readText()
+                val err = p.errorStream.bufferedReader().readText()
+                OcResult(p.exitValue(), out, err)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)